test/commutative/comparison.jl

```diff
@@ -159,4 +159,14 @@
             # Springer Science & Business Media, **2013**.
         end
     end
+
+    @testset "Comparison with NaN" begin
+        Mod.@polyvar x
+        @test isequal(NaN * x, NaN * x)
+        @test isequal(NaN * x + 0, NaN * x + 0)
+        @test isequal(NaN * x, NaN * x + 0)
+        @test !=(NaN * x, NaN * x)
+        @test !=(NaN * x + 0, NaN * x + 0)
+        @test !=(NaN * x, NaN * x + 0)
```

Comment:
Same comment for DynamicPolynomials, could you check that it's allocation free as well ?

Reply:
Fixed in https://github.com/JuliaAlgebra/MultivariatePolynomials.jl/pull/334/commits/6d3e7f32b734ac0dd1805660cfabcecbcaab8d41